### ISSUE TITLE
Nextjs Build Failure: Missing stage

### ISSUE THREAD

**Issue by hisham-otofacts** [sst/sst#3152]
## Problem

When using Github Actions to build the SST Next app with the `--stage` CLI option:

```
- name: Build application
  run: pnpm sst build --stage production
```

The following error occurs:

```
Run pnpm sst build --stage production
Attention: SST now collects completely anonymous telemetry regarding usage. This is used to guide SST's roadmap.
You can learn more, including how to opt-out of this anonymous program, by heading over to:
https://docs.sst.dev/anonymous-telemetry

Next.js v13.4.6
OpenNext v1.4.0

┌─────────────────────────────────┐
│ OpenNext — Building Next.js app │
└─────────────────────────────────┘


> otovrs@0.1.0 build /home/runner/work/otovrs-sst/otovrs-sst
> sst bind next build

Please enter a name you’d like to use for your personal stage. Or hit enter to use runner: 
┌──────────────────────────────┐
│ OpenNext — Generating bundle │
└──────────────────────────────┘

Bundling assets...
node:fs:2894
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, copyfile '/home/runner/work/otovrs-sst/otovrs-sst/.next/BUILD_ID' -> '.open-next/assets/BUILD_ID'
    at Object.copyFileSync (node:fs:2894:3)
    at createAssets (file:///home/runner/.npm/_npx/1855c526e0f0ea74/node_modules/open-next/dist/build.js:308:8)
    at build (file:///home/runner/.npm/_npx/1855c526e0f0ea74/node_modules/open-next/dist/build.js:26:5) {
  errno: -2,
  syscall: 'copyfile',
  code: 'ENOENT',
  path: '/home/runner/work/otovrs-sst/otovrs-sst/.next/BUILD_ID',
  dest: '.open-next/assets/BUILD_ID'
}
```

Notice that the log is prompting for the personal stage to be entered, even though it's being passed in as a CLI option (`pnpm sst build --stage production`) which indicates that the stage option isn't being propagated through to child commands. Because of this, it doesn't seem like the Next build is being executed which then causes the bundler to fail since the files it needs to copy doesn't yet exist.

## Workaround

As a temporary workaround, I've added a CI step to create the .sst/stage file (which I have git ignored in my repo) and to populate it with the "production" stage name.

Here's the update Github Actions code snippet:

```
- name: Build application
  run: |
    mkdir .sst
    echo "production" > .sst/stage
    pnpm sst build --stage production
```

## Proposal

The global `--stage` CLI option should be carried through to all child SST commands.

**Comment by khuezy** [comment]
I was able to pipe the stage through but not sure if this is ideal/correct.
In open next: add stage to https://github.com/serverless-stack/open-next/blob/main/packages/open-next/src/index.ts#L13
In NextjsSite: add the `-- --stage <stage>` to https://github.com/serverless-stack/sst/blob/master/packages/sst/src/constructs/SsrSite.ts#L659

**Comment by jayair** [comment]
Related Discord thread https://discord.com/channels/983865673656705025/1132312292449595553/1132371888421740664

**Comment by hisham-otofacts** [comment]
The fix should be include in version >= 2.22.10. And I've confirmed that upgrading to version 2.23.3 has fixed the issue.
